src/File.php

```diff
@@ -893,6 +893,9 @@ public function getAbsoluteURL()
      */
     public function getURL($grant = true)
```

Comment:
The acceptance criteria in the issue say this should not automatically grant by default - same as `getSourceURL()`.

Reply:
There is some discussion (https://github.com/silverstripe/silverstripe-assets/issues/501#issuecomment-1207497520) in the parent issue regarding best approach . 
If I understood properly this :
`The second option - checking permissions before providing a session grant - seems like a good default as it preserves the intent of the original implementation while clearing out bad cases.` 
I left argument `$grant` as `true` and check does user have a permission to the file or not. I also mentioned this in my comment here https://github.com/silverstripe/silverstripe-assets/issues/501#issuecomment-1248874115

Reply:
Must be `$grant = false` - the entire purpose of this original issue is to change the default of the optional param to `false`

Reply:
DONE